CarpetX/src/driver.hxx

```diff
@@ -152,13 +152,85 @@ struct GHExt {
   struct GlobalData {
     // all data that exists on all levels
 
+    class AnyTypeVector {
+    private:
+      int _type;
+      void *_data;
+      size_t _count;
+
+    public:
+      AnyTypeVector() : _type(-1), _data(nullptr), _count(0){};
+      AnyTypeVector(int type_, size_t count_)
+          : _type(type_),
+            _data(amrex::The_Arena()->alloc(CCTK_VarTypeSize(type_) * count_)),
```

Comment:
This looks dangerous. Passing an invalid `type` makes `CCTK_VarTypeSize` return a negative number, and C++ will convert this into a very large unsigned number for `alloc`. Better to check and call `CCTK_ERROR`.

Reply:
Moved into constructor code  and / or added an `assert` for the correct values. Those who compile with `NDEBUG` are on their own. In 4fc6ceb